types/applications/dialog.d.ts

```diff
@@ -0,0 +1,330 @@
+   * @param event - The keydown event
+   */
+  protected _onKeyDown(event: KeyboardEvent & { key: 'Escape' }): Promise<void>;
+  protected _onKeyDown(event: KeyboardEvent): Promise<void> | void;
+
+  /* -------------------------------------------- */
+
+  /**
+   * Submit the Dialog by selecting one of its buttons
+   * @param button - The configuration of the chosen button
+   */
+  protected submit(button: DialogButton): void;
+
+  /* -------------------------------------------- */
+
+  /**
+   * @override
+   */
+  close(options: Application.CloseOptions): Promise<void>;
+
+  /* -------------------------------------------- */
+  /*  Factory Methods                             */
+  /* -------------------------------------------- */
+
+  /**
+   * A helper factory method to create simple confirmation dialog windows which consist of simple yes/no prompts.
+   * If you require more flexibility, a custom Dialog instance is preferred.
+   *
+   * @param title       - The confirmation window title
+   * @param content     - The confirmation message
+   * @param yes         - Callback function upon yes
+   * @param no          - Callback function upon no
+   * @param render      - A function to call when the dialog is rendered
+   * @param defaultYes  - Make "yes" the default choice?
+   *                      (default: `true`)
+   * @param rejectClose - Reject the Promise if the Dialog is closed without making a choice.
+   *                      (default: `false`)
+   * @param options     - Additional rendering options passed to the Dialog
+   *                      (default: `{}`)
+   *
+   * @returns A promise which resolves once the user makes a choice or closes the window
+   *
+   * @example
+   * ```typescript
+   * let d = Dialog.confirm({
+   *  title: "A Yes or No Question",
+   *  content: "<p>Choose wisely.</p>",
+   *  yes: () => console.log("You chose ... wisely"),
+   *  no: () => console.log("You chose ... poorly"),
+   *  defaultYes: false
+   * });
+   * ```
+   */
+  static confirm<Y = true, N = false, R extends boolean = false>({
+    title,
+    content,
+    yes,
+    no,
+    render,
+    defaultYes,
+    rejectClose,
+    options
+  }?: {
+    title: string;
+    content: string;
+    yes?: (html: JQuery) => Y;
+    no?: (html: JQuery) => N;
+    render: Function;
+    defaultYes?: boolean;
+    rejectClose: R;
+    options?: Partial<Dialog.Options> & { jQuery: true };
+  }): Promise<R extends true ? Y | N : Y | N | null>;
+  static confirm<Y = true, N = false, R extends boolean = false>({
+    title,
+    content,
+    yes,
+    no,
+    render,
+    defaultYes,
+    rejectClose,
+    options
+  }?: {
+    title: string;
+    content: string;
+    yes?: (html: HTMLElement) => Y;
+    no?: (html: HTMLElement) => N;
+    render: Function;
+    defaultYes?: boolean;
+    rejectClose: R;
+    options?: Partial<Dialog.Options> & { jQuery: false };
+  }): Promise<R extends true ? Y | N : Y | N | null>;
+  static confirm({
```

Comment:
What is the purpose of the 3rd overload, which returns the `Promise<any>`? Is it to catch the case where you don't know whether `jQeury` is `true` or `false` at compile time? If so, wouldn't it make more sense to require callbacks that can take both `JQuery` and `HTMLElement` as parameter (instead of generic `Function`s) and return the proper promise (instead of `Promise<any>`)?

Reply:
Yeah you are right, I can improve that in a similar way to the other factory method. And with all that complexity also add some tests.